### ISSUE TITLE
Trailing spaces within examples in Sec. 2.2.1

### ISSUE THREAD

**Issue by notZaki** [rstudio/bookdown#545]
Hi. 

I was wondering why there are trailing spaces in the sample code in [Section 2.2.1](https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html#equations) of the book - specifically, the space after `\begin{equation}` and `\end{equation}`. 
It seems like they were [deliberately added](https://github.com/rstudio/bookdown/commit/82aed3e65d50ff6643b175b493a2518a392e3831) but I couldn't decipher the commit message. 

The reason this came up is because the sample code fails to run using `bookdown::word_document2` - a document is produced, but the equations do not show up (not an issue for html or pdf output).
This is a quirk of Word output - there can't be anything after `...{equation}` as mentioned in #190 - and I was able to get the equation to show by removing the trailing spaces.
But I'm still left wondering why the trailing space was there in the first place.

Here's sample code with trailing spaces which doesn't render properly.
```
---
output:
  bookdown::word_document2
---

\begin{equation} 
  f\left(k\right) = \binom{n}{k} p^k\left(1-p\right)^{n-k}
  (\#eq:binom) 
\end{equation} 

Equation \@ref(eq:binom) is the probability mass function of the binomial distribution with size parameter $n$ and success probability $p$. 
```

**Comment by yihui** [comment]
Yes, there was a reason, but it is a little difficult to explain: https://github.com/jgm/pandoc/issues/2758 For EPUB/Word output, bookdown has to wrap the equations inside `$$ $$`, because Pandoc does not support them out of the box like in other output formats such as LaTeX or HTML.

Then I have a problem: when I need to show verbatim examples of equations inside fenced code blocks, I need a way to tell which equations to wrap. The hidden rule in bookdown is that an equation must start with `\begin{equation}` and end with `\end{equation}` exactly:
https://github.com/rstudio/bookdown/blob/06bd6193d95ff26b4683c2fb2ce1ac3d66b3359e/R/ebook.R#L190-L208

I added trailing spaces to the examples so that they won't be wrapped in `$$ $$` when converted to EPUB/Word. Of course, this is totally hackish, and you seem to be the first person who has noticed it.

In short, for most users, there must be no leading or trailing spaces around the equation environments.

**Comment by notZaki** [comment]
Ok, so bookdown doesn't recognize the code block and wraps it in `$$...$$` anyways.
That means if I render my above sample again, but have the equation in a code block, then the word output displays the equation's code but with `$$...$$` wrapping it.
 
I imagine someone will eventually run into an issue where dollar signs mysterious start appearing in their verbatim code - hopefully they find this issue and it clears it up for them.

Thanks for the explanation, and for this package.
